src/common.ts

```diff
@@ -88,6 +88,7 @@ export class ContractService<T extends ChainId, E = Record<string, unknown>> ext
 
     try {
       const address = await this.addressProvider.addressById(contractId);
+      this.addressProvider.setCachedAddressById(contractId, address);
```

Comment:
Each service needs to call the setter because we await here for the address.

Reply:
No longer the case with https://github.com/yearn/yearn-sdk/commit/1fa4d16681b27c6b30fd1dbb3f45aa237c77f636